internal/security-governance-system/webapp/src/main/java/org/wso2/security/tools/sgs/webapp/service/ScanService.java

```diff
@@ -0,0 +1,233 @@
+
+        try {
+            URI uri = (new URIBuilder())
+                    .setHost(sgsWebappConfiguration.getScanManagerHost())
+                    .setPort(sgsWebappConfiguration.getScanManagerPort())
+                    .setScheme(SCHEME).setPath(sgsWebappConfiguration.getStartScanURI())
+                    .build();
+
+            HttpPost httppost = new HttpPost(uri);
+            MultipartEntityBuilder multipartEntityBuilder = MultipartEntityBuilder.create();
+
+            for (Map.Entry<String, MultipartFile> entry : multipartFileMultiValueMap
+                    .toSingleValueMap().entrySet()) {
+                multipartEntityBuilder.addBinaryBody(entry.getKey(), entry.getValue()
+                        .getInputStream(), ContentType.MULTIPART_FORM_DATA, entry.getValue().getOriginalFilename());
+            }
+
+            for (Map.Entry<String, String[]> entry : parameterMap.entrySet()) {
+                multipartEntityBuilder.addTextBody(entry.getKey(), convertStringArrayToString(entry.getValue(), ","));
+            }
+
+            HttpEntity httpEntity = multipartEntityBuilder.build();
+            httppost.setEntity(httpEntity);
+            client = HttpClients.createDefault();
+            HttpResponse httpResponse = client.execute(httppost);
+            responseStatus = httpResponse.getStatusLine().getStatusCode();
+
+            if (responseStatus != HttpStatus.SC_OK) {
+                logger.error("Error occurred while initiating the scan. " + responseStatus + " "
+                        + httpResponse.getStatusLine().getReasonPhrase());
+            }
+        } catch (NumberFormatException e) {
+            logger.error("Unable to retrieve the scan manager port", e);
+        } catch (URISyntaxException e) {
+            logger.error("Unable to build the start scan URI", e);
+        } catch (IOException e) {
+            logger.error("Unable to start the scan", e);
+        } finally {
+            try {
+                if (client != null) {
+                    client.close();
+                }
+            } catch (IOException e) {
+                logger.error("Unable to close the http connection with scan manager", e);
+            }
+        }
+        return responseStatus;
+    }
+
+    private static String convertStringArrayToString(String[] strArr, String delimiter) {
+        StringBuilder sb = new StringBuilder();
+        for (String str : strArr)
+            sb.append(str).append(delimiter);
+        return sb.substring(0, sb.length() - 1);
+    }
+
+    /**
+     * Get the list of scans in the pool
+     *
+     * @return
+     */
+    public ArrayList<Scan> getScans() {
+
+        MultiValueMap headerMap = new LinkedMultiValueMap();
+        Map<String, Object> paramMap = new HashMap<>();
+        ArrayList<Scan> scanList = new ArrayList<>();
+        HttpRequestHandler httpRequestHandler = new HttpRequestHandler();
+
+        try {
+            URI uri = (new URIBuilder())
```

Comment:
This URL building can be done inside the configuration class itself. That class could contain a getter that returns the required URI.